scripts/migration/migrate-addon

```diff
@@ -203,7 +241,90 @@ EOF
+  # The operator probably won't actually update Istiod faster than we will manually below, but this ensures it
+  # will not be reverted back
+  kube patch -n istio-system  istiooperators.install.istio.io istio-1-6-11-gke-0 --type=merge -p='{"spec":{
+    "components":{"pilot":{"hub":"gcr.io/gke-release/istio","tag":"1.6.14-gke.7"}}}
+  }'
+
+  # Next we insert the mesh ca root cert so that it is trusted
+  orig="$(kube -n istio-system get secret istio-ca-secret -ojsonpath='{.data.ca-cert\.pem}' | base64 -d)"
+  encoded="$(echo "${orig}" "${MESH_CA_ROOT}" | base64 -w 0)"
+  kube -n istio-system patch secret istio-ca-secret -p="{\"data\":{\"ca-cert.pem\":\"${encoded}\"}}"
+
+  # We need to restart Istiod to pick up the new image, the TD alias, and the new root cert.
+  # Patch the new image, in case operator isn't fast enough, and insert an annotation to make sure we are change something
+  kube patch deploy istiod-istio-1611 -n istio-system -p='{"spec":{"template":{"spec":{"containers":[{
+    "name":"discovery",
+    "image":"gcr.io/gke-release/istio/pilot:1.6.14-gke.7"
+  }]}}},
+  "metadata":{"annotations":{"istio-addon-migrate-start":"true"}}}'
+  kube rollout status -w deployment/istiod-istio-1611 -n istio-system
+
+  # Next, we need to convince pods to send new CSR's. Our options are to restart all pods, wait 12-24hrs,
+  # or trick Envoy to send a new SDS request. We will do the latter
+  # We add `workloadSelector: {}` because 1.4 doesn't support istio.io/rev filtering, but we don't want
+  # this to apply to 1.4 proxies or they will NACK. Instead, we abuse the fact that 1.4 rejected it,
+  # which will cause pilot to silently drop the config.
+  kube apply -f - -n istio-system << EOF
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: trigger-root-cert
+  namespace: istio-system
+  labels:
+    istio.io/rev: istio-1611
+spec:
+  workloadSelector: {}
+  configPatches:
+  - applyTo: CLUSTER
+    match:
+      context: SIDECAR_OUTBOUND
+    patch:
+      operation: ADD
+      value:
+        name: "trigger-root-cert"
+        type: STATIC
+        connect_timeout: 0.5s
+        lb_policy: ROUND_ROBIN
+        load_assignment:
+          cluster_name: trigger-root-cert
+          endpoints: []
+        transportSocket:
+          name: envoy.transport_sockets.tls
+          typedConfig:
+            '@type': type.googleapis.com/envoy.api.v2.auth.UpstreamTlsContext
+            common_tls_context:
+              validation_context_sds_secret_config:
+                name: trigger-root-cert
+                sdsConfig:
+                  initialFetchTimeout: 1s
+                  apiConfigSource:
+                    apiType: GRPC
+                    grpcServices:
+                    - envoyGrpc:
+                        clusterName: sds-grpc
+EOF
+  echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+  kube delete envoyfilter trigger-root-cert -n istio-system
```

Comment:
I guess there's not an easy way to actually check?

Reply:
Unfortunately not really. We can go exec into every pod but... not ideal